pkg/kubelet/apis/cri/runtime/v1alpha2/api.proto

```diff
@@ -597,11 +597,21 @@ message LinuxContainerConfig {
     LinuxContainerSecurityContext security_context = 2;
 }
 
+// WindowsContainerSecurityContext holds windows security configuration that will be applied to a container.
+message WindowsContainerSecurityContext {
+    // User name to run the container process as. If specified, the user MUST
+    // exist in the container image and be resolved there by the runtime;
+    // otherwise, the runtime MUST return error.
+    string run_as_username = 1;
```

Comment:
Is the username the canonical identifier in windows? If so, are there predefined significant names? Does a username on the host map to the same user & permissions as a username in the container?

We use the UID rather than username in linux for security reasons, but I'm not familiar enough with windows to know if the username is secure in this context.

Reply:
The actual canonical identifier is a Windows Security Identifier (SID), which can be looked up based on a username by the Windows Security Account Manager (SAM). The container has a separate SAM instance with a separate database from the host, therefore a user "Patrick" in the container would have a sid != that of the host. That's why the OCI spec uses a username so that it can be resolved in the context of the container. The host can't canonicalize a username to a SID without the container already running. If the username can't be canonicalized to a valid SID in the container, then creating the container's entrypoint will fail and the container will terminate.

There are some well-known ones:
Administrator - this isn't used by default in the container. Instead, …
ContainerAdministrator - this is a separate account and separate SID used in containers that doesn't exist on the host
LocalSystem, Network Service, - used by background and system processes

Reply:
Thanks for the explanation. Doesn't this mean that it's basically impossible to enforce any policy around this (at the control-plane level)? E.e. PodSecurityPolicy?

Reply:
RunAsUser, SupplementalGroups - these couldn't be implemented on Windows due to the dependency on *nix UID/GID. Windows also doesn't implement privileged or host namespace sharing (ipc, pid, net) - they're always separate. Linux CAP_* also doesn't apply.

I'll go ahead and open a blanket issue for implementing PSP on Windows.